client-spark/spark2/src/main/java/org/apache/spark/shuffle/RssShuffleManager.java

```diff
@@ -203,12 +206,12 @@ public RssShuffleManager(SparkConf sparkConf, boolean isDriver) {
   @Override
   public <K, V, C> ShuffleHandle registerShuffle(int shuffleId, int numMaps, ShuffleDependency<K, V, C> dependency) {
     // If yarn enable retry ApplicationMaster, appId will be not unique and shuffle data will be incorrect,
-    // appId + timestamp can avoid such problem,
+    // appId + uuid can avoid such problem,
```

Comment:
Why do we need uuid?

Reply:
We can't get the appId when we try Access, because the appId is generated after the RssManager is created. In order to support push down, we maintain the uuid as a substitute for the appId, and replace the uuid with the appId after the app heartbeat is reported to the coordinator.

Reply:
We can put the appId to the `AcessInfo` when we try to access coordinator.

Reply:
Do you mean to generate the uuid on the driver?

Reply:
Do you mean to generate the uuid on the driver?

Reply:
My fault. We ignore that I can't get the appId in the construct method. Let me think twice.